packages/@aws-cdk/aws-lambda/lib/function.ts

```diff
@@ -672,7 +677,7 @@ export class Function extends FunctionBase {
     if (props.architectures && props.architectures.length > 1) {
       throw new Error('Only one architecture must be specified.');
     }
-    const architecture = props.architecture ?? (props.architectures && props.architectures[0]);
+    const architecture = props.architecture ?? (props.architectures && props.architectures[0]) ?? Architecture.X86_64;
```

Comment:
Oh no, this is not what I meant by my [previous comment](https://github.com/aws/aws-cdk/pull/17121/files#r736272136). I meant only about fixing the getter. 
The change you have now will cause every CDK user's stack to change (undesirable).

However, I just decided in a [different PR](https://github.com/aws/aws-cdk/pull/17000#discussion_r739116360) to leave such getters as `undefined` if the user has not supplied it. Let's revert this back to returning `undefined` if not specified.

Apologies for the churn.